### DISCUSSION TITLE
Widget fadeout relative to time until habit is non-chainable

### DISCUSSION THREAD

**Discussion by NegatioN** [iSoron/uhabits#597] [Feature Requests]
Preface: Love the app.

For me the widget (my main interface) is not enough in my face about when a habit is no longer chainable, since it's simply in a state of `Checked`(completely lit up) or `Unchecked` (Black background / Transparent).

A long time ago I used another app (MyChain, now somewhat defunct) which had a widget progressively fading out (or moving progressively from Green-->Red) which I found really effective in bringing "keeping the habit going" to the front of my mind when I was about to slack off.

I'm a horrible illustrator, but I think simply scaling the alpha of the uhabits widget-color as a function of `(maxHabitGapTime - timeSinceDone)/maxHabitGapTime` would work similarly well for this purpose.

I see the potential negative in introducing a new pattern not everyone might enjoy in the same way though. Interested to hear your thoughts on this. And keep up the good work :slightly_smiling_face:

**Comment by Waog** [comment]
I suggested something similar. #829

**Reply by sh0nuff** [reply]
I think that a lot of the other similar ideas aren't the same as this one.. Having a widget change from green to yellow to orange to red as the day progresses increases pressure to complete certain tasks. This could also be connected to "do task x times a day" where the color starts as grey/unchecked, and with each press the color changes from red to orange to yellow to green as you approach the target
